third_party/amd/lib/TritonAMDGPUTransforms/Utility.h

```diff
@@ -16,4 +18,13 @@ using namespace mlir;
 int deduceMinCountOnDefChain(Value defValue, Operation *consumerOp,
                              llvm::function_ref<int(Operation *)> countFunc);
 
+// Returns a padded shared encoding minimizing bank conflicts for the given
+// tensor and dot encoding.
+std::optional<triton::gpu::PaddedSharedEncodingAttr>
+composePaddedLayout(const triton::AMD::TargetInfo &targetInfo,
+                    triton::gpu::DotOperandEncodingAttr dotOpEnc,
+                    triton::gpu::TensorOrMemDesc srcTy,
+                    ArrayRef<unsigned> order, ArrayRef<unsigned> sharedOrder,
+                    unsigned bitWidth, bool useAsyncCopy);
```

Comment:
Some of the parameters like `order`, `bitWidth` can be queried from `srcTy` to avoid divergence? Do we need to expose them?